### PR TITLE
add performance -- infra metrics and loss curves

### PR DESCRIPTION
**Pull request by tianyu-l**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #237

WPS / MFU numbers, and loss curves jobs can be found from this tracking [spreadsheet](https://docs.google.com/spreadsheets/d/11kcula5ybuABSZkm2OlFng5NQ9_rnVB-KRyeQq6P7fo/edit#gid=0).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,18 +18,20 @@
 
 Please note: `torchtitan` is a proof-of-concept for Large-scale LLM training using native PyTorch. It is (and will continue to be) a repo to showcase PyTorch's latest distributed training features in a clean, minimal codebase. torchtitan is complementary to and not a replacement for any of the great large-scale LLM training codebases such as Megatron, Megablocks, LLM Foundry, Deepspeed, etc. Instead, we hope that the features showcased in torchtitan will be adopted by these codebases quickly. torchtitan is unlikely to ever grow a large community around it.
 
+
 ## Pre-Release Updates:
-#### (4/16/2024): TorchTitan is now public but in a pre-release state and under development.  Currently we showcase pre-training Llama2 models (LLMs) of various sizes from scratch.
+#### (4/16/2024): `torchtitan` is now public but in a pre-release state and under development.  Currently we showcase pre-training Llama2 models (LLMs) of various sizes from scratch. `torchtitan` is tested and verified with the PyTorch nightly version `torch-2.4.0.dev20240412`.
```
Comment (lessw2020) [inline README.md:23]: My only concern is that stating a fixed version might pull us into providing backward support based on claiming fixed version.  At a minimum, imo we should add "We recommend using the latest nightlies" or similar to avoid people running this fixed version 3 months from now but then filing issues on things not working that are otherwise fixed in latest nightly.

Comment (gnadathur) [reply README.md:23]: ok -- I think this was because of the discussions we had. Maybe the language can be altered to " we recommend using PyTorch nightly version `torch-2.4.0.dev20240412` or later"
